src/object.c

```diff
@@ -1812,3 +1803,89 @@ char *object_get_name(rct_object_entry *entry)
 
 	return (char *)pos;
 }
+
+void *object_aligned_alloc_pad(size_t alignment, size_t size)
+{
+	int remainder = size % alignment;
+	size_t zero_pad = (remainder > 0) ? (alignment - remainder) : 0;
+
+	// Allocate enough space for ...:
+	//  - the alignment
+	//  - a pointer to the original return value (needed for free)
+	//  - the object
+	//  - extra zero-padded space to bring object size to a multiple of 32
+	//  - extra space for alignment
+	void *ptr = (uint8 *)malloc(sizeof(size_t) + sizeof(uint8 *) + size + alignment - 1 + zero_pad);
+	if (ptr == NULL) {
+		log_error("Failed to allocate memory for object.");
+		assert(false);
+	}
+
+	// Align chunk pointer on boundary.
+	void *ptr_aligned = (void *)((uint8 *)ptr + sizeof(size_t) + sizeof(uint8 *) + alignment - 1 - (size % alignment));
+
+	// Save alignment for realloc
+	((size_t *)ptr_aligned)[-2] = alignment;
```

Comment:
Curiously enough, the implementation in MinGW seems to use the same method as I implemented. The libc versions work the same too, but they can free up the header and extra padding since they have direct control over memory allocation.